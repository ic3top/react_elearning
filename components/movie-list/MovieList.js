import { connect } from 'react-redux';
import { useFetchMovies } from '/hooks/useFetchMovies';

import { MovieCard } from '../movie-card/MovieCard';

import styles from './MovieList.module.scss';

export function MovieList({ movies, amount }) {
  const { setSearch } = useFetchMovies();

  const content = movies?.length
    ? movies.map((movie) => (
      <MovieCard
        key={movie?.id}
        movie={movie}
        onClick={() => setSearch({ movie: movie?.id })}
      />
    ))
    : (
      <p>
        Sorry, we couldn&apos;t find anything :(
        <style jsx>{'p { color: white; font-size: 24px  }'}</style>
      </p>
    );

  return (
    <>
      <p className={styles.moviesAmount}>
        <span>{amount || 0}</span>
        {' '}
        movies found
      </p>
      <div className={styles.movies}>
        {content}
      </div>
    </>
  );
}

const mapStateToProps = (state) => ({
  movies: state.movies.value,
  amount: state.movies.totalAmount,
});

export default connect(mapStateToProps)(MovieList);
