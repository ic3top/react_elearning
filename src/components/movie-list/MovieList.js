import {useSearchParams} from "react-router-dom";
import {connect} from "react-redux";

import './movie-list.scss';
import {MovieCard} from "../movie-card/MovieCard";

export const MovieList = ({ movies, amount }) => {
  const [search, setSearch] =  useSearchParams();

  return (
    <>
      <p className="movies__found"><span>{amount}</span> movies found</p>
      <div className="movies">
        {movies.map(movie =>
          <MovieCard
            key={movie.id}
            movie={movie}
            onClick={() => setSearch({ ...Object.fromEntries(search), movie: movie.id }) }
          />
        )}
      </div>
    </>
  )
}

const mapStateToProps = (state, props) => ({
    movies: state.movies.value,
    amount: state.movies.totalAmount,
})

export default connect(mapStateToProps)(MovieList);
