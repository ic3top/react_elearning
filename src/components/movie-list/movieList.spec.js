import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { MovieList } from './MovieList';

describe('MovieList', () => {
    const movies = [{
            release_date: 'release_date',
            title: 'title',
            genres: ['Action'],
            poster_path: 'poster_path',
            id: 'id1'
        },
        {
            release_date: 'release_date',
            title: 'title',
            genres: ['Action'],
            poster_path: 'poster_path',
            id: 'id2'
        },
    ];

    let mockStore;

    beforeEach(() => {
      mockStore = configureStore();
    });

  it('renders correctly all movies', () => {
    const tree = renderer.create(
         <Provider store={mockStore({})}>
            <Router>
                <MovieList movies={movies} amount={movies.length} />
            </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
