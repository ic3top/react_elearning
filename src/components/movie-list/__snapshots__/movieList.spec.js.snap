// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders correctly all movies 1`] = `
Array [
  <p
    className="movies__found"
  >
    <span>
      2
    </span>
     movies found
  </p>,
  <div
    className="movies"
  >
    <div
      className="movie-card"
      onMouseLeave={[Function]}
      role="listitem"
    >
      <div
        className="movie-card__img-wrapper"
      >
        <div
          className="movie-card__actions"
        >
          <button
            onClick={[Function]}
          >
            <img
              alt="three dots"
              src="dots-more.png"
            />
          </button>
        </div>
        <div>
          <div
            className="movie-card__modal false"
          >
            <button
              className="movie-card__close"
              onClick={[Function]}
            >
              <img
                alt="x-mark"
                src="close-sm.png"
              />
            </button>
            <button
              className="movie-card__btn"
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              className="movie-card__btn"
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
        </div>
        <img
          alt="movie"
          className="movie-card__img"
          onClick={[Function]}
          onError={[Function]}
          src="poster_path"
        />
      </div>
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__title"
        >
          title
        </div>
        <div
          className="movie-card__year"
        >
          release_date
        </div>
      </div>
      <div
        className="movie-card__genre"
      >
        Action
      </div>
    </div>
    <div
      className="movie-card"
      onMouseLeave={[Function]}
      role="listitem"
    >
      <div
        className="movie-card__img-wrapper"
      >
        <div
          className="movie-card__actions"
        >
          <button
            onClick={[Function]}
          >
            <img
              alt="three dots"
              src="dots-more.png"
            />
          </button>
        </div>
        <div>
          <div
            className="movie-card__modal false"
          >
            <button
              className="movie-card__close"
              onClick={[Function]}
            >
              <img
                alt="x-mark"
                src="close-sm.png"
              />
            </button>
            <button
              className="movie-card__btn"
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              className="movie-card__btn"
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
        </div>
        <img
          alt="movie"
          className="movie-card__img"
          onClick={[Function]}
          onError={[Function]}
          src="poster_path"
        />
      </div>
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__title"
        >
          title
        </div>
        <div
          className="movie-card__year"
        >
          release_date
        </div>
      </div>
      <div
        className="movie-card__genre"
      >
        Action
      </div>
    </div>
  </div>,
]
`;
